nimble/host: Add support for storing client supported features values

piotrnarajowski opened the pull request:
Implements the core mechanism for storing and maintaining Client Supported Features for peers with a trusted relationship. This successfully resolves Host qualification test case GATT/SR/GAS/BV-04-C.

Alongside the core stack changes, storage layer has been extended with a new object type, and a comprehensive unit test suite (ble_gatts_cl_sup_feat_suite) has been added

piotrnarajowski committed:
nimble/host: Add support for new type of object in the storage.

Add support for BLE_STORE_OBJ_TYPE_PEER_CL_SUP_FEAT. This is used
for storing values of Client Supported Features for clients with a
trusted relationship.

piotrnarajowski committed:
nimble/host: maintain Client Supported Features characteristic value

This adds support for storing client supported features for
clients with a trusted relationship. Fixes Host qualification test
case GATT/SR/GAS/BV-04-C

piotrnarajowski commented:
#AutoPTS run mynewt GATT/SR/GAS/BV-04-C

piotrnarajowski commented:
#AutoPTS run mynewt GATT/SR/GAS/BV-04-C

piotrnarajowski committed:
nimble/host: update storage unit tests

Extend storage unit tests to cover new object type PEER_CL_SUP_FEAT

piotrnarajowski committed:
nimble/host: fix code style

Fix code style in ble_store.h

piotrnarajowski committed:
nimble/host: move BLE_GATT_CHR_CLI_SUP_FEAT_SZ to public header.

Moves 'BLE_GATT_CHR_CLI_SUP_FEAT_SZ' from the private header to the
public ble_gatt.h header.

piotrnarajowski commented on nimble/host/include/host/ble_store.h:
```diff
@@ -29,6 +29,7 @@
 
 #include <inttypes.h>
 #include "nimble/ble.h"
+#include "../../host/src/ble_gatt_priv.h"
```
moved to `ble_gatt.h`

piotrnarajowski committed:
nimble/host: Add unit tests for maintaining Client Supported Features

Introduce a comprehensive unit test suite to validate the persistance,
restoration, and specification compliance of the Client Supported
Features characteristic values across peer bonding lifecycles.

piotrnarajowski commented:
@MariuszSkamra PTAL

piotrnarajowski commented:
#AutoPTS run mynewt GATT/SR/GAS/BV-04-C

codecoup-tester commented:
Scheduled PR https://github.com/apache/mynewt-nimble/pull/2114#issuecomment-4612790146, board: nrf52, estimated start time: 15:27:54, test case count: 1, estimated duration: 0:10:57<details><summary>Test cases to be run</summary>GATT/SR/GAS/BV-04-C</details>

codecoup-tester commented:
AutoPTS Bot results:
No failed test found.
<details><summary>Successful tests (1)</summary>GATT    GATT/SR/GAS/BV-04-C             PASS</details>

piotrnarajowski committed:
apps/btshell: update keystore commmands

Update keystore commands to handle Client Supported Features
storage type object.

piotrnarajowski commented:
Recent changes introduce Btshell keystore methods update that allows to see if Client Supported Features values has been stored properly for client with trusted relationship
